### DISCUSSION TITLE
Cannot read property 'digest' of undefined

### DISCUSSION THREAD

**Discussion by akinwol** [supabase/supabase#17101] [Questions]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

I am not sure what I am doing wrong but I am stuck on sign up on react native. This is on localhost. 

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

My imports 

`import { useSupabaseClient } from "@supabase/auth-helpers-react";`

```
  const supabase = useSupabaseClient();
 const { data, error } = await supabase.auth.signUp({
        email: formData.email,
        password: formData.password,
        options: {
          data: {
            full_name: formData.name,
          },
        },
      });

```

I keep getting this error ` {"error": [TypeError: Cannot read property 'digest' of undefined]}`

What's more strange is that sign in works fine if I create the user manually from the supabase dashboard. 

## Expected behavior

On signup im expecting a user to be successfully signed up and to get an email notification with the confirmation code 

## System information

- OS: macOS
- Version of supabase-js: 
```
    "@supabase/auth-helpers-react": "^0.4.2",
    "@supabase/supabase-js": "^2.33.1",
```

## Additional context
- expo version: ` "expo": "^49.0.7",`
- react native version: `react-native": "0.72.4",`

**Comment by saltcod** [comment]
@akinwol converted this to a troubleshooting Discussion. Discord would also be a good place to debug this one.

**Comment by itsmnjn** [comment]
can confirm am facing the same issue

`"react-native": "0.71"`
`"expo": "^48.0.17"`

**Reply by itsmnjn** [reply]
seems like it's due to gotrue using `crypto.subtle` library which can only be used in certain environments

https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest

**Comment by GoldAndLink** [comment]
Same issue while calling `supabase.auth.signInWithOtp` with magic link

`"react-native": "0.72.4"`
`"expo": "~49.0.8"`

Any updates ?
